packages/core/src/WorkflowExecute.ts

```diff
@@ -305,6 +314,81 @@ export class WorkflowExecute {
 		return this.processRunExecutionData(workflow);
 	}
 
+	// IMPORTANT: Do not add "async" to this function, it will then convert the
+	//            PCancelable to a regular Promise and does so not allow canceling
+	//            active executions anymore
+	// eslint-disable-next-line @typescript-eslint/promise-function-async
+	runPartialWorkflow2(
+		workflow: Workflow,
+		runData: IRunData,
+		_startNodes: StartNodeData[],
+		destinationNodeName?: string,
+		pinData?: IPinData,
+	): PCancelable<IRun> {
+		if (destinationNodeName === undefined) {
+			throw new ApplicationError('destinationNodeName is undefined');
```

Comment:
You and me both.

So right now this comes from the FE. If you press this button it won't send the `destinationNode`
![image](https://github.com/user-attachments/assets/061cbec1-5e52-4f87-8f5b-57f90c62e020)

And if we press the play button on a node, it will send the node name as the destination node.
The contract between FE and BE is a bit weird though:
In the old flow you could send `runData` and no destination node. It would just execute from the start node until the end. But the FE never made use of this.
In the new flow I require the `destinationNodeName` to be sent and also to exist in the workflow. So if these assertions fail, it's a programmer error in my opinion.

How about I replace these `ApplicationError`s with simple asserts?

Reply:
Sounds good to me!